### ISSUE TITLE
error trying to promote float infinity to bigfloat in sign comparison

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2277]
Imported from SourceForge on 2024-07-05 22:40:35
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-11-13 22:39:56
Original: https://sourceforge.net/p/maxima/bugs/3244

---

This continues work started in #3235. 

With current source code + GCL 2.6.12, I get:

~~~~
$ sh maxima-local -l gcl
Maxima branch_5_38_base_348_g47c7d24_dirty http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) assume(x>200b0);
(%o1)                             [x > 2.0b2]
(%i2) limit(x*exp(x)*log(exp(x)+1),x,-1000,plus);

bfloat: attempted conversion of floating-point infinity.
~~~~

It's reported that the last two tests in rtest_limit fail with that error.

It appears the problem is that there is a bigfloat in the assume database, and DCOMPARE is attempting to compare negative float infinity to it, and tries to promote the float infinity to bigfloat, and gets the error. 

I'll assume here that it's OK for there to be a float infinity somewhere and that the problem is trying to promote float infinity to bigfloat.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:37
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2016-11-15 09:22:44
Original: https://sourceforge.net/p/maxima/bugs/3244/#4b36

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:40
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2016-11-15 09:22:44
Original: https://sourceforge.net/p/maxima/bugs/3244/#f121

---

I don't think that it's OK for there to be a float infinity in maxima, since there
are tests to prevent it using FLOAT-INF-P in various places.  I have added a few
more calls to FLOAT-INF-P to resolve this bug in commit [06b469f6f8ef3c05d65ef9fb5e8fbacaeafb30c4].

A better way to fix this would have been to enable trap on floating point 
overflow in gcl and ecl, but I can't get that working at the moment.

In gcl, adding
~~~~
     (eval-when (load eval)
       (si::break-on-floating-point-exceptions :floating-point-overflow t))
~~~~
to the maxima sources doesn't actually turn on the overflow trap.
Issuing the same command to lisp from the maxima command line does
work and does resolve the error in the test suite.

In ecl, adding
~~~~
(si::trap-fpe 'floating-point-overflow t)
~~~~
to the maxima sources does turn on the overflow trap and does resolve
the error in the test suite, but triggers an error in slatec.  I get a nondeterministic
overflow error in computing the bessel function:
~~~~
(%i5) for a:0 thru 2000 do bessel_j(-2.3,1.0+%i);

Maxima encountered a Lisp error:

 #<a FLOATING-POINT-OVERFLOW>
~~~~
This error in computing the bessel function can be resolved by initializing the
arrays cwrkr and cwkri in file zuoik.lisp.  I'm not sure whether it makes sense 
to modifiy f2cl to initialize the arrays, or to report this as a bug to ecl.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:44
Created by **[dimpase](https://sourceforge.net/u/dimpase/)** on 2016-11-15 16:18:06
Original: https://sourceforge.net/p/maxima/bugs/3244/#f121/2cd7

---

Which ECL version are you working with?

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:48
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2016-11-15 16:38:44
Original: https://sourceforge.net/p/maxima/bugs/3244/#f121/2cd7/dcf0

---

ECL 16.1.2

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:51
Created by **[dimpase](https://sourceforge.net/u/dimpase/)** on 2016-11-15 16:45:51
Original: https://sourceforge.net/p/maxima/bugs/3244/#f121/2cd7/dcf0/f47d

---

They have updated their floating point stack (and added infinities) following #3235, see
https://gitlab.com/embeddable-common-lisp/ecl/issues/299
(which also fixes the original issue of #3235 on 5.38.1 built with ECL)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 22:40:55
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2016-11-15 17:03:10
Original: https://sourceforge.net/p/maxima/bugs/3244/#f121/2cd7/3147

---

On Tue, November 15, 2016 at  4:18PM, Dima Pasechnik wrote:
> Which ECL version are you working with?

ECL 16.1.2

Dan
